data-model/src/main/java/io/micronaut/data/model/query/builder/AbstractSqlLikeQueryBuilder.java

```diff
@@ -1091,13 +1104,23 @@ protected void appendOrder(QueryModel query, QueryState queryState) {
             StringBuilder buff = queryState.getQuery();
             buff.append(ORDER_BY_CLAUSE);
             Iterator<Sort.Order> i = orders.iterator();
+
+            AnnotationValue<JsonView> jsonViewAnnotationValue = queryState.getEntity().getAnnotationMetadata().getAnnotation(JsonView.class);
+            String jsonViewColumnName = null;
```

Comment:
This should be reading the annotation metadata of the repository method and check for `QueryResult`

Reply:
Yes, I think you are right. But wasn't sure how to annotate with QueryResult on the fly based on JsonView annotation. I guess it should be done in visitor somehow? Or some other way?

Reply:
Probably in `RepositoryTypeElementVisitor#visitMethod` just before we start processing the method

Reply:
I have added this change, please have a look.